app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java

```diff
@@ -124,55 +128,59 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
+        ContentResolver content = getContentResolver();
         MimeTypeMap mime = MimeTypeMap.getSingleton();
-        return mime.getExtensionFromMimeType(c.getType(contentUri));
+        return mime.getExtensionFromMimeType(content.getType(contentUri));
     }
     
-    //creating collision resistant file name.This method creates a unique file name for new photo using data time stamp
-    //recommended method of creating image file from the camera
+    //This method creates a unique file name for new photo using data time stamp
+    //This method is used when photoFile is null
     private File createImageFile() throws IOException {
-        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());//creates time stamp
-        String imageFileName = ";JPEG_" + timeStamp + "_";//file name starts with JPEG followed by timestamp
-        // File storageDir = getExternalFilesDir(Environment.DIRECTORY_PICTURES) to save our file 
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());  //creates time stamp
+        String imageFileName = ";JPEG_" + timeStamp + "_";  //creating image file
+        // Standard directory to place image
         File storageDir = getExternalFilesDir(Environment.DIRECTORY_PICTURES);
-        File image = File.createTempFile(//creating image file using the method of CreateTempFile we are passing image parameters
+        //creating image file using method of CreateTempFile
+        File image = File.createTempFile(
                 imageFileName, /* prefix */
                 ".jpg", /* suffix */
                 storageDir /* directory */
         );
-        // Save a file: path for use with ACTION_VIEW intents
-        currentPhotoPath = image.getAbsolutePath();//using AbsolutePath of image ,image can be displayed in imageview
+        // gets absolute path of image
+        currentPhotoPath = image.getAbsolutePath();
         return image;
     }
-    //open the camera and save our image file into the directory
+
+    //this method opens the camera and save our image file into the directory
     private void dispatchTakePictureIntent() {
         Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
-        // Ensure that there is a camera activity to handle the intent , whether camera permission is given to the activity
+        // ensures camera is ready
         if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
             // Create the File where the photo should go
             File photoFile = null;
+            //createImageFile method within try catch block to prevent IOException
             try {
-                photoFile = createImageFile();//createImageFile can throw IOException so put inside try catch block.Creating photoFile assigning to Create image file which returns image
+                photoFile = createImageFile();  //this returns image
             } catch (IOException ex) {
```

Comment:
Do not keep unhandled exception! Always throw an error message or handle the exception (if there is any other possibility to solve the exception)